### ISSUE TITLE
Geeksofwaregmbh.pdf24creator fetching older version than displayed from winget output

### ISSUE THREAD

**Issue by Karl-WE** [microsoft/winget-pkgs#92590]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

Upgrade of package geeksofwaregmbh.pdf24creator will show Upgrade from 11.8.0 to 11.9.0 correctly, yet download 11.3.0 

### Steps to reproduce

Have Pdf24 installed in a not latest version. Use exe or msi from vendors homepage.

Winget Upgrade  --all

### Actual behavior

An outdated Version 11.3.0 will be downloaded from the vendors server, indicated by filename and effective version that's being installed. 

### Expected behavior

Winget should fetch and install the version 11.9.0

### Environment

```shell
Windows-Paket-Manager (Vorschau) v1.4.3132-preview
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Windows: Windows.Desktop v10.0.25267.1000
Systemarchitektur: X64
Paket: Microsoft.DesktopAppInstaller v1.19.3132.0

Protokolle: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
```

### Screenshots and Logs

![Screenshot_20221229_170830](https://user-images.githubusercontent.com/45657752/210051024-fa7f9588-4643-4837-aee7-cf05c67cd443.png)

**Comment by mdanish-kh** [comment]
Looks like the issue is caused by InstallerType differences. The current version on your machine seems to have been installed through an inno-based `exe` installer but the latest manifest only contained `msi`. I've pushed a PR that'll hopefully fix this issue.

**Comment by mdanish-kh** [comment]
@Karl-WE Fixed at my end; Can you please confirm?

```pwsh
~ winget install geeksoftwareGmbH.PDF24Creator --version 10.8.0
Found PDF24 Creator [geeksoftwareGmbH.PDF24Creator] Version 10.8.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://stx.pdf24.org/products/pdf-creator/download/pdf24-creator-10.8.0.exe
  ██████████████████████████████   127 MB /  127 MB
Successfully verified installer hash
Starting package install...
Successfully installed
```
```pwsh
~ winget list pdf24
Name                 Id                            Version Available Source
---------------------------------------------------------------------------
PDF24 Creator 10.8.0 geeksoftwareGmbH.PDF24Creator 10.8.0  11.9.0    winget
```
```pwsh
~ winget upgrade pdf24
Found PDF24 Creator [geeksoftwareGmbH.PDF24Creator] Version 11.9.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://download1.pdf24.org/pdf24-creator-11.9.0.exe
  █▉                              18.0 MB /  285 MB
Package download canceled.
```

**Comment by Karl-WE** [comment]
Thank you for this prompt investigation.

Tried the following as a repro

- install 11.7.0 exe (inno based) from https://www.pdf24.org/listversions.php archived versions

- run winget upgrade --all
- reported version is correctly determined
- lastest version is correctly determined
- winget will DL and install 11.9.0 successfully
- double checked installed pdf24.exe version  to report 11.9.0

All fine.

- run winget uninstall pdf24 (terminated background Service)
- repeat installing msi based setup of 11.7.0
- all results as expected

Great job!

This issue is solved.

One question aside: will yaml files be updated via AI or manually as I often had reliabilty issues with the availability of new versions showing up in winget community repo. It's a very popular tool in Germany, so I hope you don't mind asking me this.

**Comment by mdanish-kh** [comment]
As it stands, there is some form of automation available through @wingetbot but this package and most others have to be manually updated and reviewed. The issue here was that Inno installer was included in previous manifests but recently it was removed in a community submission and unfortunately it went unnoticed. I'll be sure to thoroughly review to prevent this from happening if I come across a PR for this package.

**Comment by Karl-WE** [comment]
Thanks! I would be interested what would be required to enable wingetbot to take over.
Since the vendor offers a universal webpage listing all versions, SHA and filenames this sounds like a good start for automation as they take this product very seriously and update it frequently.

**Comment by mdanish-kh** [comment]
You can take a look at https://github.com/microsoft/winget-pkgs/issues/500 to see what has been considered up until now.

**Comment by borque** [comment]
Hello!
As I am the one responsible for the exe-removal let me explain this:
As @Karl-WE said, this tool is very popular in Germany but the manifests here were not updated for months.
So I decided to "jump in" and create some updated manifests but due to a lack of time and bandwith (I did not know the SHA256 are provided by the vendor) I only included the MSI-installers.
As I ran into the same problem with Thunderbird and Firefox.ESR I started a discussion here:  #92746

**Comment by Karl-WE** [comment]
I can confirm it has not been updated a long time I believe stuck on 11.3.0 when you updated it. Thanks for that. I had a misconception about this has not to be done manually. I know better now.

**Comment by mdanish-kh** [comment]
Close with reason: Resolved in #92611;

**Comment by elreymon** [comment]
Hi:

Installation of pdf24 through winget got stuck (freezed). I had to close the cmd window. I think it´s because in normall installation the installer checks if the pdf24 service is running in the background and asks de user to stop it. As Winget does not shows this window, the installer gets stuck.

I will investigate further and give more info. The service is used to have a pdf virtual printer. Thanks anyway.
